### PR TITLE
Improve chainerx import check

### LINKED ISSUE #7730: Improve chainerx import check
#7518 has disabled the following:
1. Check out the source code and `import chainer` without pip install.
2. pip install chainer in non-editable mode (with/without chainerx) and import chainer from the source root directory.

\1. should be supported. In 2., we should give more comprehensible error.

### PR DESCRIPTION
**Pull request by asi1024**
Fix #7730.
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/imanishi/chainer/chainerx/__init__.py", line 10, in <module>
    '$ pip install -e {}'.format(root_dir))
ImportError: Reinstall Chainer in "editable" mode.
$ pip install -e /home/imanishi/chainer
```

### CODE REVIEW

**Comment by niboshi** [comment]
It may be helpful for users if more detailed information is provided in the message.
The information we want to include is:


>Cannot import chainerx because `_build_info.py` cannot be found.
>It may be caused by either
>1. you are directly importing chainer source files without installing it with `pip install`, or
>2. you installed chainer in non-editable mode (`pip install` without `-e`) and importing chainer source files instead of the installed module.
>For 1, please install chainer using pip in editable mode.
>For 2, please avoid importing source files if chainer is installed in non-editable mode.

There seems to be a room for better organization of the information in the message above.

**Comment by asi1024** [comment]
@niboshi PTAL.

File: chainerx/__init__.py [hunk 1]
```diff
@@ -1,8 +1,23 @@
 import os
 import warnings
 
-from chainerx import _build_info
-
+try:
+    from chainerx import _build_info
+except ImportError:
+    chainer_root_dir = os.path.abspath(__file__ + '../../..')
```
Comment (niboshi) [inline chainerx/__init__.py]: This variable is not used.

File: chainerx/__init__.py [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+    raise ImportError(
+        '''\
+Cannot import chainerx because _build_info.py cannot be found.
+
+The chainerx module being imported was not correctly installed by \
```
Comment (niboshi) [inline chainerx/__init__.py]: How about "chainer and chainerx"?
Those who `import chainer` will also be affected, and they may have difiiculty interpret the message if it only mentions `chainerx`.

**Comment by niboshi** [comment]
Jenkins, test this please

**Pull request merged**